File: solutions/go/etl/1/etl.go
package etl

import (
	"strings"
)

func Transform(in map[int][]string) map[string]int {
	r := make(map[string]int)
	for score, letters := range in {
		for _, letter := range letters {
			letter = strings.ToLower(letter)
			r[letter] = score
		}
	}
	return r
}

File: solutions/haskell/etl/1/src/ETL.hs
module ETL (transform) where

import Data.Char (toLower)
import Data.Map (Map, assocs, fromList)
import Data.Text (Text, unpack)

transform :: Map a Text -> Map Char a
transform legacyData =
    fromList
        [ (toLower letter, a)
        | (a, letters) <- assocs legacyData
        , letter <- unpack letters
        ]
